ovirt_imageio/client/_options.py

```diff
@@ -10,62 +10,124 @@
 Tool options.
 """
 
+from collections import namedtuple
+
 from .. _internal.units import KiB, MiB, GiB, TiB
 
 import argparse
 import configparser
 import os
+import sys
+
+
+class Option(
+        namedtuple("Option", "name,args,config,required,type,default,help")):
+
+    __slots__ = ()
+
+    def __new__(cls, name=None, args=(), config=False, required=False,
+                type=str, default=None, help=None):
+        return tuple.__new__(
+            cls, (name, args, config, required, type, default, help))
+
+
+def log_level(s):
+    if s not in ("debug", "info", "warning", "error"):
+        raise ValueError(f"Invalid 'log_level' value: '{s}'")
+    return s
 
 
 class Parser:
 
+    _OPTIONS = [
+        Option(
+            name="config",
+            args=["-c", "--config"],
+            help="If set, read specified section from the configuration file.",
+        ),
+        Option(
+            name="engine_url",
+            args=["--engine-url"],
+            config=True,
+            required=True,
+            help=("ovirt-engine URL. If not set, read from the specified "
+                  "config section (required)."),
+        ),
+        Option(
+            name="username",
+            args=["--username"],
+            config=True,
+            required=True,
+            help=("ovirt-engine username. If not set, read from the "
+                  "specified config section (required)."),
+        ),
+        Option(
+            name="password",
+            config=True,
+        ),
+        Option(
+            name="password_file",
+            args=["--password-file"],
+            help=("Read ovirt-engine password from file. If not set, read "
+                  "password from the specified config section."),
+        ),
+        Option(
+            name="cafile",
+            args=["--cafile"],
+            config=True,
+            help=("Path to ovirt-engine CA certificate. If not set, read "
+                  "from the specified config section"),
+        ),
+        Option(
+            name="log_file",
+            args=["--log-file"],
+            config=True,
+            help="If set, log to file.",
+        ),
+        Option(
+            name="log_level",
+            args=["--log-level"],
+            config=True,
+            type=log_level,
+            default="info",
+            help="Logging level (default: info).",
```

Comment:
Since log_level without log_file (which has no default) is useless, should that combination be an error? Or should we log to stdout?

Reply:
We set the log level only when the log file is set, but actually the logger is using filename=stderr and level=warning by default. I think we can make this more general by setting the log level in all case. If you use --log-level=debug your will get very verbose output in the shell.

Reply:
Log level is always used now, and the default value is warning.

Example usage with default log level:
```
$ ./ovirt-img download-disk -c engine adb9b2c7-32a9-4e87-894c-710de7165086 \
    /data/fc/tmp/dl.qcow2
[ 100% ] 6.00 GiB, 1.85 seconds, 3.24 GiB/s                                    
```

With info log level:
```
$ ./ovirt-img download-disk -c engine adb9b2c7-32a9-4e87-894c-710de7165086 \
    /data/fc/tmp/dl.qcow2 --log-level info
2022-07-27 01:19:00,643 INFO    (MainThread) [ovirt] Creating transfer image=Disk(id='adb9b2c7-32a9-4e87-894c-710de7165086') direction=download host=None backup=None shallow=None timeout_policy=cancel
2022-07-27 01:19:01,205 INFO    (MainThread) [ovirt] Transfer 'ed600506-61ed-4bfe-afb9-76662e7cd551' ready on host 'host4' in 0.6 seconds
[ 100% ] 6.00 GiB, 1.86 seconds, 3.22 GiB/s                                    
2022-07-27 01:19:03,067 INFO    (MainThread) [ovirt] Finalizing transfer 'ed600506-61ed-4bfe-afb9-76662e7cd551'
2022-07-27 01:19:05,103 INFO    (MainThread) [ovirt] Transfer 'ed600506-61ed-4bfe-afb9-76662e7cd551' finalized in 2.0 seconds
```